### ISSUE TITLE
Streamlined font install and config update

### ISSUE THREAD

**Issue by svrooij** [JanDeDobbeleer/oh-my-posh#2338]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What would you like to see changed/added?

I really like **oh-my-posh**, but on a new PC I'm always struggling with the following stuff:

- Picking the right font and the right font version
- Downloading the correct ZIP
- Installing the font
- Changing the config in all the apps I would like to use the new font.

Maybe this can be automated during installation possibly with two commanline switches:

- `--install-font {name-font}` with sensible default off-course
- `--config-font {vscode,terminal,vs}` so `--config-font` with a list of apps that are supported for auto configuring the font.

These commands can maybe also be baked into a new cli for managing dev fonts, what do you think?

**Comment by JanDeDobbeleer** [comment]
@svrooij this is a very good use-case to implement a [bubbletea](https://github.com/charmbracelet/bubbletea) style setup command that will guide you through this (where we can). It would mean a few things though.

- store the list of fonts in oh-my-posh
- limit to terminals that have a config file stored on disk (terminal/vscode)

**Comment by svrooij** [comment]
Are you suggesting a separate cli tool or as part of oh-my-posh? My go knowledge is like none existing, so I would require some pointers on how to start and how to build/debug it.

**Comment by JanDeDobbeleer** [comment]
@svrooij part of it. The hardest part will be figuring out how install a font using the CLI. Never really did that before but there's examples.

**Comment by JanDeDobbeleer** [comment]
@svrooij to install fonts, I found [this](https://github.com/Crosse/font-install) one. It can even install from a zip file which means we can fetch all available fonts, download and install using the logic in this codebase (it's a bit outdated so it needs an update).

**Comment by JanDeDobbeleer** [comment]
<img width="1249" alt="image" src="https://user-images.githubusercontent.com/2492783/171417000-9692f005-93c9-40e3-af5c-4306e307b67a.png">

**Comment by ehawman-rosenberg** [comment]
@JanDeDobbeleer ooh that is slick. For clarity, is this installing the font system-wide, or just for OMP's use?

**Comment by svrooij** [comment]
@ehawman-rosenberg I hope globally.

I talked to Jan about this feature, and the goal is to allow the quick installation of the needed fonts ☑️ and configure those fonts as the configured font for the supported terminals (Windows terminal, vscode, visual studio dev terminal, possibly others).

**Comment by JanDeDobbeleer** [comment]
@ehawman-rosenberg it's system wide. I'll have this out in the coming week for sure. Then we can see how to deal with configuring the terminal afterwards. Although it might need one go as the font name needs to be correct. So we can chain these commands, I don't feel like also querying all installed fonts (unless that trivial, don't know just yet).

**Comment by JanDeDobbeleer** [comment]
@svrooij I would be honoured if you could try out the install functionality (be aware as we install system wide, this needs to be run as admin): `oh-my-posh font install`

**Comment by ehawman-rosenberg** [comment]
@JanDeDobbeleer tested and working on Mac. Beautiful stuff.

**Comment by ehawman-rosenberg** [comment]
@JanDeDobbeleer is it possible to pass a NF name directly to the command, bypassing the bubbletea user session? I have an Unraid server that wipes itself on reboot and requires a bootstrapping script to reinstall some component parts. While this is a rare occurrence, it would be cool to include something like `oh-my-posh font install CascadiaCode` in my bootup script and have that take place without my input (it runs as root so no auth issues).

**Comment by JanDeDobbeleer** [comment]
@ehawman-rosenberg can do. Would need to be an exact match but yes.

**Comment by JanDeDobbeleer** [comment]
@ehawman-rosenberg with the change above, you can use any valid font name (as defined in the [release](https://github.com/ryanoasis/nerd-fonts/releases/tag/v2.1.0) page without `.zip`):

```bash
oh-my-posh font install CascadiaCode
```

EDIT: removed `--font`

**Comment by ehawman-rosenberg** [comment]
@JanDeDobbeleer Not sure if it's just my Windows system at work but that isn't working.
![image](https://user-images.githubusercontent.com/81652082/172404898-0ae5e132-9405-46be-a469-ef8a06a7aa5c.png)  
Note the admin indicator in the top left.

![image](https://user-images.githubusercontent.com/81652082/172405074-24883f6b-b226-434b-8b07-f21a6ecc09ce.png)

Pretty sure debug won't do you any good, but here ya go

<details>

```bash
Version: 8.3.1

Segments:

ConsoleTitle(true)   -   0 ms - ~
iterm(false)         -   0 ms -
os(true)             -   0 ms -  ═
shell(true)          -   0 ms -   nu
session(true)        -   1 ms -  ehawman@EHSupportLaptop
angular(false)       -   0 ms -
aws(false)           -   0 ms -
az(false)            -   0 ms -
azfunc(false)        -   0 ms -
go(false)            -   0 ms -
java(false)          -   0 ms -
julia(false)         -   0 ms -
kubectl(false)       - 158 ms -
node(false)          -   0 ms -
php(false)           -   0 ms -
project(false)       -   0 ms -
python(false)        -   0 ms -
ruby(false)          -   0 ms -
time(true)           -   0 ms -  Tue | 09:25:08
executiontime(true)  -   0 ms -  0ms 
git(false)           -  10 ms -
text(true)           -   0 ms -  ╚
exit(true)           -   0 ms -  
path(true)           -   1 ms -  
text(true)           -   0 ms -   

Run duration: 197.238ms

Cache path: C:\Users\ehawman\AppData\Local\oh-my-posh

Config path: C:\Users\ehawman\AppData\Local\Programs\oh-my-posh\themes\eh_custom\devious-diamonds.omp.yaml

Logs:

2022/06/07 09:25:08 debug: Getenv
C:\Users\ehawman\AppData\Local
2022/06/07 09:25:08 Getenv duration: 0s, args: LOCALAPPDATA
2022/06/07 09:25:08 CachePath duration: 0s, args:
2022/06/07 09:25:08 debug: Getenv
C:\Users\ehawman\AppData\Local\Programs\oh-my-posh\themes\eh_custom\devious-diamonds.omp.yaml
2022/06/07 09:25:08 Getenv duration: 0s, args: POSH_THEME
2022/06/07 09:25:08 Shell duration: 18.9636ms, args:
2022/06/07 09:25:08 resolveConfigPath duration: 18.9636ms, args:
2022/06/07 09:25:08 Init duration: 19.473ms, args:
2022/06/07 09:25:08 Flags duration: 0s, args:
2022/06/07 09:25:08 config.loadConfig duration: 1.0493ms, args:
2022/06/07 09:25:08 Flags duration: 0s, args:
2022/06/07 09:25:08 debug: Getenv

2022/06/07 09:25:08 Getenv duration: 0s, args: OMP_CACHE_DISABLED
2022/06/07 09:25:08 WindowsRegistryKeyValue duration: 0s, args: HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 debug: WindowsRegistryKeyValue
ColorizationColor(DWORD): 0xC4204355
2022/06/07 09:25:08 Root duration: 0s, args:
2022/06/07 09:25:08 Shell duration: 0s, args:
2022/06/07 09:25:08 ErrorCode duration: 0s, args:
2022/06/07 09:25:08 IsWsl duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 debug: User
ehawman
2022/06/07 09:25:08 User duration: 0s, args:
2022/06/07 09:25:08 debug: Host
EHSupportLaptop
2022/06/07 09:25:08 Host duration: 507.5µs, args:
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 507.5µs, args:
2022/06/07 09:25:08 Flags duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Getenv

2022/06/07 09:25:08 Getenv duration: 0s, args: TERM_PROGRAM
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 Shell duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 Shell duration: 0s, args:
2022/06/07 09:25:08 Flags duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 Shell duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Getenv

2022/06/07 09:25:08 Getenv duration: 0s, args: SSH_CONNECTION
2022/06/07 09:25:08 debug: Getenv

2022/06/07 09:25:08 Getenv duration: 0s, args: SSH_CLIENT
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 Shell duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Getenv

2022/06/07 09:25:08 Getenv duration: 0s, args: AWS_VAULT
2022/06/07 09:25:08 debug: Getenv

2022/06/07 09:25:08 Getenv duration: 0s, args: AWS_PROFILE
2022/06/07 09:25:08 debug: Getenv

2022/06/07 09:25:08 Getenv duration: 0s, args: AWS_REGION
2022/06/07 09:25:08 debug: Getenv

2022/06/07 09:25:08 Getenv duration: 0s, args: AWS_DEFAULT_REGION
2022/06/07 09:25:08 debug: Getenv

2022/06/07 09:25:08 Getenv duration: 0s, args: AWS_CONFIG_FILE
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 error: FileContent
open C:\Users\ehawman/.aws/config: The system cannot find the path specified.
2022/06/07 09:25:08 FileContent duration: 0s, args: C:\Users\ehawman/.aws/config
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Getenv

2022/06/07 09:25:08 Getenv duration: 0s, args: AZURE_CONFIG_DIR
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 debug: HasFolder
false
2022/06/07 09:25:08 HasFolder duration: 0s, args: C:\Users\ehawman\.azure
2022/06/07 09:25:08 debug: HasFolder
false
2022/06/07 09:25:08 HasFolder duration: 0s, args: C:\Users\ehawman\.Azure
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 debug: Getenv

2022/06/07 09:25:08 Getenv duration: 0s, args: JAVA_HOME
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 HasCommand duration: 12.9428ms, args: kubectl
2022/06/07 09:25:08 error: RunCommand
cmd.Start() failed with 'error: current-context must exist in order to minify
'
2022/06/07 09:25:08 RunCommand duration: 145.575ms, args: kubectl config view --output yaml --minify
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 debug: HasFiles
false
2022/06/07 09:25:08 HasFiles duration: 0s, args: package.json
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 debug: HasFiles
false
2022/06/07 09:25:08 HasFiles duration: 0s, args: Cargo.toml
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 debug: HasFiles
false
2022/06/07 09:25:08 HasFiles duration: 0s, args: pyproject.toml
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 debug: HasFiles
false
2022/06/07 09:25:08 HasFiles duration: 0s, args: composer.json
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 debug: HasFiles
false
2022/06/07 09:25:08 HasFiles duration: 749.5µs, args: *.nuspec
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 debug: HasFiles
false
2022/06/07 09:25:08 HasFiles duration: 267.8µs, args: *.py
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 debug: HasFiles
false
2022/06/07 09:25:08 HasFiles duration: 213.5µs, args: *.ipynb
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 debug: HasFiles
false
2022/06/07 09:25:08 HasFiles duration: 0s, args: pyproject.toml
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 debug: HasFiles
false
2022/06/07 09:25:08 HasFiles duration: 0s, args: venv.bak
2022/06/07 09:25:08 debug: HasFolder
false
2022/06/07 09:25:08 HasFolder duration: 0s, args: .venv
2022/06/07 09:25:08 debug: HasFolder
false
2022/06/07 09:25:08 HasFolder duration: 0s, args: venv
2022/06/07 09:25:08 debug: HasFolder
false
2022/06/07 09:25:08 HasFolder duration: 0s, args: virtualenv
2022/06/07 09:25:08 debug: HasFolder
false
2022/06/07 09:25:08 HasFolder duration: 0s, args: env
2022/06/07 09:25:08 debug: HasFolder
false
2022/06/07 09:25:08 HasFolder duration: 0s, args: venv-win
2022/06/07 09:25:08 debug: HasFolder
false
2022/06/07 09:25:08 HasFolder duration: 0s, args: pyenv-win
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 Shell duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 ExecutionTime duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 Shell duration: 0s, args:
2022/06/07 09:25:08 Flags duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 HasCommand duration: 10.4671ms, args: git.exe
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 error: HasParentFilePath
CreateFile C:\.git: The system cannot find the file specified.
2022/06/07 09:25:08 HasParentFilePath duration: 0s, args: .git
2022/06/07 09:25:08 Flags duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 Shell duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 ErrorCode duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 Shell duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 Flags duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 Flags duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 PathSeparator duration: 0s, args:
2022/06/07 09:25:08 IsWsl duration: 0s, args:
2022/06/07 09:25:08 StackCount duration: 0s, args:
2022/06/07 09:25:08 DirIsWritable duration: 46.2µs, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 Shell duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 debug: Pwd
C:\Users\ehawman
2022/06/07 09:25:08 Pwd duration: 0s, args:
2022/06/07 09:25:08 debug: Home
C:\Users\ehawman
2022/06/07 09:25:08 GOOS duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 Shell duration: 0s, args:
2022/06/07 09:25:08 TemplateCache duration: 0s, args:
2022/06/07 09:25:08 debug: Getenv
C:\Users\ehawman\AppData\Local
2022/06/07 09:25:08 Getenv duration: 0s, args: LOCALAPPDATA
2022/06/07 09:25:08 CachePath duration: 0s, args:
2022/06/07 09:25:08 Flags duration: 0s, args:
```
</details>

EDIT: It's able to determine a proper download or not 
![image](https://user-images.githubusercontent.com/81652082/172406123-ff6ddefc-8e78-469a-aea6-875d69e10433.png)

EDIT: Tested on Powershell, no joy.

**Comment by JanDeDobbeleer** [comment]
@ehawman-rosenberg did you reboot? This requires a registry reload.

**Comment by ehawman-rosenberg** [comment]
@JanDeDobbeleer Ahaha I did not. I'll try that later.

**Comment by JanDeDobbeleer** [comment]
@ehawman-rosenberg I just checked Windows, while it reports as successful, it does not seem to have the font. Something for later I guess.

**Comment by ehawman-rosenberg** [comment]
@JanDeDobbeleer Just rebooted and checked. No joy here either.

I tested the Bubbletea process on Windows and it didn't work for me either, even after a reboot. Weeeird.

**Comment by JanDeDobbeleer** [comment]
@ehawman-rosenberg it seems that you need to [call a win32 API](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/nf-wingdi-addfontresourcea) for this to work. So only copying the file and adjusting the registry doesn't work. I'll try to see if I can adjust the code to use this.

**Comment by JanDeDobbeleer** [comment]
@ehawman-rosenberg I think I found the solution, it was quite the work to get this to where it needed to be. Can you validate?

**Comment by ehawman-rosenberg** [comment]
@JanDeDobbeleer

>it was quite the work to get this to where it needed to be

Isn't that always the case with "simple little" ideas? Sometimes it feels like the apparent complexity of a task is inversely proportional to the amount of frustrated effort you'll need to put in to solve it.

As for testing, it seems the update hasn't quite made it through the winget pipeline yet. I'm going to lunch and hopefully it'll be ready when I get back. I'll let you know soon :)

**Comment by ehawman** [comment]
@JanDeDobbeleer 4 things

1. Hi! This is my personal account, which I probably should have been using the whole time but whatever. I'm going to try to use this one from now on.
2. I had to install manually rather than wait on winget. Typically winget is ready to go mere minutes after you post a release, so maybe check on the process to see if anything has gone wrong?
3. It works perfectly on Windows 10. Mission Accomplished
4. Some user feedback would be appreciated. Something as simple as "{font} installed sucessfully!" would go a long way. Bringing back the stuff from this image would be even better 
![image](https://user-images.githubusercontent.com/52979227/172715480-f9cb3d2b-919e-476d-8cf9-0121e6143787.png)

**Comment by svrooij** [comment]
It seems to add `.zip` if I use the sample command provided by oh-my-posh. So the download now ends in `.zip.zip` which fails.
![image](https://user-images.githubusercontent.com/1292510/172788026-47afeee9-3682-4014-9355-1899fc5a1490.png)

**Comment by JanDeDobbeleer** [comment]
@svrooij you only need the name, not the URL. There's an example above 😃

**Comment by svrooij** [comment]
@JanDeDobbeleer I know, but `oh-my-posh font -h` suggests the command above.

![image](https://user-images.githubusercontent.com/1292510/172790151-2b8e7b31-ef95-47cc-9473-c7995e1d77ee.png)


I tried picking a font from the list, and now it still moving but I dont' feel it's doing much. Like this for 2 minutes now.
![image](https://user-images.githubusercontent.com/1292510/172790100-07e1c3a1-419d-4491-8b59-ae95d6dbb09b.png)

**Comment by seffyroff** [comment]
I'm seeing the same thing as @svrooij here.  Clean install of OMP via winget.  
![image](https://user-images.githubusercontent.com/8338529/173130969-462f4b7c-1a5a-45ed-a569-c8b2ac8a5f14.png)

**Comment by JanDeDobbeleer** [comment]
I'll have a look, could be a slow connection (or GitHub is blocked). I've ran this code at least 100 times already without failure...

**Comment by seffyroff** [comment]
In trying to get you more info, I've watched what it's doing, and it seems to just be stuck at the end after installing for me. If I have the Windows fonts dialog open I can see the new font appear as the command is running.  The command never completes however.

On top of that, I think what I was actually seeing in the case of UbuntuMono is that the font is already installed anyway.

**Comment by JanDeDobbeleer** [comment]
@seffyroff could be that one of the win32 calls is stuck (or fails and then the UI doesn't update). I'll check that scenario.

**Comment by JanDeDobbeleer** [comment]
@svrooij @seffyroff can you try again with the latest release? I removed a call that wasn't really required and some other minor improvements.

**Comment by JanDeDobbeleer** [comment]
@svrooij @seffyroff friendly reminder ;-)

**Comment by JanDeDobbeleer** [comment]
No new bug reports for font install and the config tweaks are out of scope (too hard to do correctly with maintainable code).

**Comment by frank1003A** [comment]
I tried this and got an error log that is obviously missing a few words
![errorfont](https://user-images.githubusercontent.com/63779386/235596843-c5c6ede2-ef5b-4e01-8856-bfbb4f5e2abe.jpg)
I also tried to install: oh-my-posh font install MesloLGM NF, and still get the zip file error.
![errorfont2](https://user-images.githubusercontent.com/63779386/235597650-d590c96b-0a7a-439a-8ee5-fdd9c9323ffb.jpg)
I don't what else I can do at this point

**Comment by JanDeDobbeleer** [comment]
@frank1003A follow #3782

**Comment by felipeoriani** [comment]
I had the similar issue trying to install fonts on Windows Terminal on Windows 10. When running `oh-my-posh font install` and selecting a font it seems to succeed instalation but after reboot windows it did not work.

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/865173/657a0b8b-67f9-4f31-98b5-bf35a21563cd)

 I was not able to set it on the Windows Terminal settings. 

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/865173/a411494c-7f76-4316-a233-783b56b69399)

Suggestions? 
Thank you.

**Comment by JanDeDobbeleer** [comment]
@felipeoriani is the font name correct?

**Comment by felipeoriani** [comment]
@JanDeDobbeleer Yes, I think everything is correct since I followed this tutorial https://ohmyposh.dev/docs/installation/fonts

If I check on C:\Windows\fonts, I can't see the `Meslo` font family, even after reboot.

Note: My Windows 10 installation is new and updated, just format my PC a few days ago.

**Comment by JanDeDobbeleer** [comment]
@felipeoriani hmm, are you on the latest version of oh-my-posh? Because we did fix something with fonts in [v16.2.3](https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v16.2.3)

**Comment by felipeoriani** [comment]
Sorry @JanDeDobbeleer , I was using the v16.2.2. Installed the v16.2.3 and installed the font again. It works fine now, but the example on the link does not work. Instead setting `"face": "MesloLGM NF"` I had to set: `"face": "MesloLGL Nerd Font"`.

Thank you.
